File: dotnet/Scenarios/Exercises/Logic/Repository/Models/SectionEntity.cs
namespace EncyclopediaGalactica.Scenarios.Exercises.Logic.Repository.Models;

public class SectionEntity
{
    public long Id { get; set; }
    public string? Title { get; set; }

    public double SectionNumber { get; set; }

    public int PageStart { get; set; }

    public int PageExercisesStart { get; set; }

    public int ConceptQuestionsIntervalStart { get; set; }

    public int ConceptQuestionsIntervalEnd { get; set; }

    public int SkillQuestionsIntervalStart { get; set; }

    public int SkillQuestionsIntervalEnd { get; set; }

    public int ApplicationQuestionsIntervalStart { get; set; }

    public int ApplicationQuestionsIntervalEnd { get; set; }

    public int DiscussionQuestionsIntervalStart { get; set; }

    public int DiscussionQuestionsIntervalEnd { get; set; }

    public int PageEnd { get; set; }
    public long ChapterId { get; set; }
    public ChapterEntity Chapter { get; set; }

    public List<ExerciseEntity> Exercises { get; set; } = [];
}

File: Rust/eg/src/logic/orm/section_entity.rs
use welds::WeldsModel;

use super::chapter_entity::ChapterEntity;

#[derive(Debug, WeldsModel)]
#[welds(table = "sections")]
#[welds(BelongsTo(chapter, ChapterEntity, "chapter_id"))]
pub struct SectionEntity {
    #[welds(primary_key)]
    pub id: i32,
    pub section_title: String,
    pub section_number: f64,
    pub page_start: i32,
    pub page_exercises_start: i32,
    pub concepts_questions_interval_start: i32,
    pub concepts_questions_interval_end: i32,
    pub skills_questions_interval_start: i32,
    pub skills_questions_interval_end: i32,
    pub applications_questions_interval_start: i32,
    pub applications_questions_interval_end: i32,
    pub discussion_questions_interval_start: i32,
    pub discussion_questions_interval_end: i32,
    pub page_end: i32,
    pub chapter_id: i32,
}
